### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by JbirdL86**
What's new :question: 

- Create a new main file
- Refactor the main method
- Refactor library methods to comply with SOLID rules

### CODE REVIEW

File: library.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    book_selection = gets.chomp.to_i
+    book = @book_list[book_selection]
+
+    puts 'Select a person from the following list by number (not ID):'
+    @people.map do |person|
+      puts "#{@people.index(person)}) [#{person.class}] Name: #{person.name}. ID: #{person.id}, Age: #{person.age}"
+    end
+
+    print 'Person #: '
+    person_selection = gets.chomp.to_i
```
Comment (SajjadAhmad14) [inline library.rb]: Please follow the `single responsibility rule` and move the `create_rental` method from the Library class to the `Rental` class.

File: main.rb [hunk 2]
```diff
@@ -1,138 +1,46 @@
...
+
+    option = gets.chomp.to_i
+
+    case option
+    when 1
+      app.list_books
+    when 2
+      app.list_people
+    when 3
+      app.people.push(app.create_person)
```
Comment (SajjadAhmad14) [inline main.rb:29]: A person can not be created when the user wants to create a student. Please resolve the issue.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi all, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: main.rb (hunk 2 above)
Comment (mateo951) [reply main.rb:29]: Done ✔

File: library.rb (hunk 1 above)
Comment (mateo951) [reply library.rb]: Done ✔

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi guys 🥇, 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**